### PR TITLE
test: Add test for ApiTraceGraph and order methods in dfs 

### PR DESCRIPTION
**Pull request by rish691**
Change order of methods 
Checkin a trace and add preliminary test 
Note: No code changes

### CODE REVIEW

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json [hunk 1]
```diff
@@ -0,0 +1,4163 @@
+{
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json:1]: Curious: Why are we storing this as well? For readability?

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraphTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    SpecificDatumReader<StructuredTrace> datumReader = new SpecificDatumReader<>(
+        StructuredTrace.getClassSchema());
+    DataFileReader<StructuredTrace> dfrStructuredTrace = new DataFileReader<>(new File(resource.getPath()), datumReader);
+
+    StructuredTrace trace = dfrStructuredTrace.next();
+    ApiTraceGraph apiTraceGraph = new ApiTraceGraph(trace);
+    apiTraceGraph.build();
+    assertTrue(apiTraceGraph.getApiNodeEventEdgeList().size() > 0);
+    assertTrue(apiTraceGraph.getNodeList().size() > 0);
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraphTest.java:29]: Can we have more concrete conditions as we have the same trace?
- exact number of nodes
- exact number of edges

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java [hunk 3]
```diff
@@ -29,14 +29,20 @@
...
 
-  private List<ApiNode<Event>> nodeList;
-  private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
   private final StructuredTrace trace;
+  private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
   private final Map<String, Integer> eventIdToIndexInTrace;
 
+  // map of api node to index in api nodes list, since we want to build edges between api nodes
+  private Map<ApiNode<Event>, Integer> apiNodeToIndex;
+  //map of entry boundary event to api node. each api node has one entry boundary event
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:42]:
```suggestion
  // map of entry boundary event to api node. each api node has one entry boundary event
```

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java [hunk 4]
```diff
@@ -29,14 +29,20 @@
...
       EnrichedSpanConstants.getValue(AttributeValue.ATTRIBUTE_VALUE_UNKNOWN);
 
-  private List<ApiNode<Event>> nodeList;
-  private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
   private final StructuredTrace trace;
+  private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
   private final Map<String, Integer> eventIdToIndexInTrace;
 
+  // map of api node to index in api nodes list, since we want to build edges between api nodes
+  private Map<ApiNode<Event>, Integer> apiNodeToIndex;
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:41]: I know it was not previously initialized, do you think, we should init in the constructor?

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json (hunk 1 above)
Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json:1]: yes

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraphTest.java (hunk 2 above)
Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraphTest.java:29]: more will come in subsequent pr

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java [hunk 5]
```diff
@@ -12,6 +13,7 @@
 import java.util.Queue;
 import java.util.Set;
 import java.util.stream.Collectors;
+import org.apache.commons.compress.utils.Lists;
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:16]:
```suggestion
```

Comment (kotharironak) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java:16]: Above is un-used import

**Review by kotharironak — APPROVED** [review]

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json (hunk 1 above)
Comment (skjindal93) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json:1]: Or, we could have read the structured trace directly from JSON in the code

Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/resources/SampleTrace-Hotrod.json:1]: Actually I tried doing that, but while converting to avro it was failing. 
I will move to using the json once I figure out why.

**Pull request merged**